### ISSUE TITLE
log package - Error functions don't marshal as expected

### ISSUE THREAD

**Issue by necrophonic** [ONSdigital/go-ns#60]
Given
```
err := errors.New("An error")
```
Using
```
log.Error(err, nil)
```
I'm expecting the output to be something like:
```
... "data": { "error": "An error", "message": "An error" } ...
```
(though it looks odd repeating the message; that's what the code seems to imply it should do)

I'm getting however:
```
... "data": { "error": {}, "message": "An error" } ...
```

I think this is to do with the way Go won't marshal interface types (which the `err` is). It's not getting picked up in the unit tests as they're looking at the pre-marshalled data which has the correct values.

**Comment by necrophonic** [comment]
Maybe I'm misinterperating what the output is supposed to be though? If the `error={}` is supposed to be dumping the error object then possibly in the event it's empty, the error key should be dropped?

**Comment by ian-kent** [comment]
yeah, this isn't right, but `Error`, `ErrorR` and `ErrorC` aren't quite fit for purpose and need to be refactored (probably a breaking change)

**Comment by necrophonic** [comment]
Related now to #62
